<a href="https://colab.research.google.com/github/prithvikolla/GitHub_Repository_Analysis/blob/main/GRA_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

import io

import requests

urlfile="https://raw.githubusercontent.com/prithvikolla/GitHub_Repository_Analysis/main/2015-01-01-0.csv"

url=requests.get(urlfile).content

df_GRA =pd.read_csv(io.StringIO(url.decode('utf-8')))


In [2]:
df_GRA.head()

id  ...                                  org.avatar_url
0  2489368070  ...                                              ""
1  2489368072  ...                                              ""
2  2489368089  ...                                              ""
3  2489368095  ...  https://avatars.githubusercontent.com/u/82592?
4  2489368104  ...                                              ""

[5 rows x 182 columns]

In [3]:
df_GRA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7702 entries, 0 to 7701
Columns: 182 entries, id to org.avatar_url
dtypes: bool(1), float64(1), int64(3), object(177)
memory usage: 10.6+ MB


In [4]:
unique_actors = list (df_GRA['actor.login'].unique())

In [5]:
len(unique_actors)

3648

In [6]:
len(df_GRA['actor.id'].unique())

3648

1. How many unique contributors for each group?

- Contributors can be inferred as , users who have done push event and finding the distinct users by "actor.id" attribute and "type" attribute.

In [7]:
id = [ ]
for i in range(len(df_GRA['type'])): 
  if (df_GRA['type'][i] == 'PushEvent'):
    id.append(df_GRA['actor.id'][i])

In [8]:
def uniq(input):
  output = [ ]
  for x in input:
    if x not in output:
      output.append(x)
  return output
id_unique = uniq(input = id)

In [9]:
len(id_unique)

2327

In [10]:
print(len(id) == len(id_unique))

False


hence, all the push events done have same user contributing multiple times

- so, number of unique contributions are 2327 by the analysis done till now.

2. What's the distribution of the number of contributors among the all projects here? 
 - what is the mean?
 - what is the standard deviation? 
 - Is it a normal distribution?

In [13]:
from nltk import Counter

In [15]:
event_Count = Counter(df_GRA['type'])

In [16]:
event_Count.most_common(3)

[('PushEvent', 4280), ('CreateEvent', 815), ('IssueCommentEvent', 650)]

As we can see,In the top 3 common events done in Github. We find that users did largely "Push Event" and the second one is "Create Event"

In [25]:
#Check wether we have entire columns in the whole data set with null data.
null_data = pd.DataFrame()
for i in df_GRA.columns:
  if (len(df_GRA[i])== 0):
    null_data.insert(0,i,df_GRA[i])
null_data.head()

Empty DataFrame
Columns: []
Index: []

In [26]:
len(null_data)
## No we do not have entire data columns in the whole data set with null values.

0